### DISCUSSION TITLE
Prioritising a peer

### DISCUSSION THREAD

**Discussion by GaryElshaw** [transmission/transmission#3596] [Ideas]
I'm not even sure this is even in a spec, but it would be great to be able to prioritise a peer.

**Comment by ile6695** [comment]
Do you mean downloading or seeding to/from specific peer?
I don't think that downloading from a specific peer would be really nice towards that peer, so I assume prioritizing seeding.
Seeding/uploading however is done when a peer requests blocks of data from seeder. In that case we can only send those blocks that were requested. Maybe a peer does not want some of the files in a torrent or is asking them from some other peer and is already saturating their bandwidth.
There could be a case where seeder has bandwidth for only some of the peers and then decides who gets the data first, but a healthy swarm needs many peers that all have some (or all) blocks of the data. Sending blocks to many directions helps the data to move faster in total, because everyone who has some blocks is able to share them.

**Comment by robd003** [comment]
It would be interesting to try to seed blocks that don't exist on other peers at a higher priority to help the swarm.
